### PR TITLE
[FIX] web_editor: align icon to the right

### PR DESCRIPTION
**Pull request by bso-odoo**
When replacing an image by a pictogram, the classes of the previous media
are copied to the new icon's `<span>`. This potentially includes the
`w-100` class which prevents the alignment from behaving correctly.

This commit removes the `w-100` class if it exists - in addition to
removing the inline `width` attribute which was done before.

Steps to reproduce:
- Drop a "Media List" block
- Replace first image by pictogram.
- Align icon to the right.
=> Icon was displayed aligned to the left.

task-2829971 (was task-2729177)

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by bso-odoo** [comment]
Only the first commit must be forward-ported, not the second one.

**Comment by bso-odoo** [comment]
@dmo-odoo Can someone from `web_editor` give a look ?
It might seem overkill to do it in two commits - but the summernote-specific one should not be forward-ported.

**Comment by dmo-odoo** [comment]
@bso-odoo I'm not sure I understand what is going on with the two commits. Since they both modify the same line, wouldn't it be better to have two PR ? One for 14.0 with one commit for 14.0 which will not be forward ported, and one for 15.0 with one commit for 15.0 that will be properly forward ported.

**Comment by bso-odoo** [comment]
The 15.0 fix will require https://github.com/odoo/odoo/pull/85600 to be merged first.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/web_editor/static/src/js/wysiwyg/widgets/media.js [hunk 1]
```diff
@@ -1066,6 +1066,9 @@ var IconWidget = SearchableMediaWidget.extend({
         if (!this.$media.is('span, i')) {
             var $span = $('<span/>');
             $span.data(this.$media.data());
+            // Classes will be restored from previous media but icons width
+            // must be unspecified.
+            this.$media.removeClass('w-25 w-50 w-75 w-100');
```
[the remark this answers is not in the document]
Comment (bso-odoo) [reply addons/web_editor/static/src/js/wysiwyg/widgets/media.js]: Indeed, moved the changes from "going to pictogram" to "leaving from image".

**Review by qsm-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Pull request closed without merging**